src/uuid/uuidbuilder.rs

```diff
@@ -33,174 +32,196 @@ pub struct UUIDBuilder {
     lsb: u64,
 }
 
-impl Default for UUIDBuilder {
-    /// Creates a new builder for creating uProtocol UUIDs.
-    ///
-    /// Simply invokes [`UUIDBuilder::new`].
-    ///
-    /// # Examples
+impl UUIDBuilder {
+    /// Builds a new UUID with consistent `rand_b` portion no matter which thread or task this is
+    /// called from. The `rand_b` portion is what uniquely identifies this uE.
     ///
-    /// ```rust
-    /// use up_rust::UUIDBuilder;
+    /// # Returns
     ///
-    /// let builder = UUIDBuilder::default();
-    /// let uuid = builder.build();
-    /// assert!(uuid.is_uprotocol_uuid());
-    /// ```
-    fn default() -> Self {
-        Self::new()
+    /// UUID with consistent `rand_b` portion, which uniquely identifies this uE
+    pub fn build() -> UUID {
+        #[allow(clippy::redundant_closure)]
```

Comment:
Maybe we can have this:
`static UUIDBUILDER_SINGLETON: Lazy<UUIDBuilder> = Lazy::new(UUIDBuilder::new);`

Reply:
Ahhh okay. Makes sense